packages/dev/core/src/Materials/Node/Blocks/storageReadBlock.ts

```diff
@@ -0,0 +1,71 @@
+import { NodeMaterialBlock } from "../nodeMaterialBlock";
+import { NodeMaterialBlockConnectionPointTypes } from "../Enums/nodeMaterialBlockConnectionPointTypes";
+import type { NodeMaterialBuildState } from "../nodeMaterialBuildState";
+import { NodeMaterialConnectionPointDirection, type NodeMaterialConnectionPoint } from "../nodeMaterialBlockConnectionPoint";
+import { NodeMaterialBlockTargets } from "../Enums/nodeMaterialBlockTargets";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { LoopBlock } from "./loopBlock";
+import { NodeMaterialConnectionPointCustomObject } from "../nodeMaterialConnectionPointCustomObject";
+/**
+ * Block used to read from a variable within a loop
+ */
+export class StorageReadBlock extends NodeMaterialBlock {

```

Comment:
`StorageReadBlock` and `StorageWriteBlock` can only be used in connection with a `Loop` block (the `loopID` input must be connected to a `Loop` block), so maybe add "Loop" somewhere in the class name to make it clear these classes work together?

Reply:
Well I'm not a big fan of over indexing with a longer name. As the ports are mandatory and linked to a loop, I feel it is enough